web/src/components/hooks/useDownloadValues.js

```diff
@@ -33,12 +33,28 @@ const useDownloadValues = ({
   appSlug,
   fileName,
 } = {}) => {
-  const ref = useRef(null);
+  // const ref = useRef(null);
+
   const [isDownloading, setIsDownloading] = useState(false);
   const [error, setError] = useState(null);
   const [url, setUrl] = useState(null);
   const [name, setName] = useState(null);
 
+  useEffect(() => {
+    if (url) {
+      const link = document.createElement("a");
+      link.href = url;
+      link.setAttribute("download", name);
+
+      document.body.appendChild(link);
+
+      link.click();
+      link.parentNode.removeChild(link);
+      _revokeObjectURL(url);
+      setUrl(null);
+    }
+  }, [url]);
+
```

Comment:
The ref was getting updated with the new url before it was updated in the dom. Decided to just do it all with synchronous dom updates rather than spreading it across lifecycle methods.